### DISCUSSION TITLE
Can I access a language client registered by an extension?

### DISCUSSION THREAD

**Discussion by invm** [CodinGame/monaco-vscode-api#483] [Q&A] [answered]
I'm running the demo with remote authority and remote path and an extension that provides language config and language server, I see the output of the language server in the outputs tab but I wonder if I can make my own requests?

I was able to do this with `monaco-langaugeclient` like this:
```tsx
const lc = wrapper.getLanguageClientWrapper();
const res = await lc
  ?.getLanguageClient()
  ?.sendRequest('textDocument/documentSymbol', {
    textDocument: { uri: 'file:///workspace/resources.tf' },
  });
console.log(res);
```

Can I access registered langugage clients?

Thanks

**Comment by CGNonofr** [comment]
The language client runs in the extension code, so there's no way to access it, except if the extension explicitly export it to other extensions running in the same extension host

**Comment by CGNonofr** [comment, marked as answer]
But instead of running the LSP request directly, you can probably do it via the VSCode api: 
```typescript
const symbols = await vscode.commands.executeCommand(
  'vscode.executeDocumentSymbolProvider',
  vscode.Uri.parse('file:///workspace/resources.tf')
) as vscode.DocumentSymbol[];
```

**Reply by invm** [reply]
Can confirm that this works as expected and is sufficient for me, thank you.
